### PR TITLE
Add a new CLI option to limit the number of requests in a single RPC batch request

### LINKED ISSUE #4951: Add a new CLI option to limit the number of requests in a single RPC batch request
### Description

Adds a `--rpc-max-batch-size` CLI option to restrict the maximum number of requests in a single RPC batch request.

Default `50`
When `-1`: unlimited - the current configuration.

When number of requests in batch > `--rpc-max-batch-size`:

- Throw a JsonRpcError.EXCEEDS_RPC_MAX_BATCH_SIZE; with type `INVALID_PARAM` and message `Number of requests in batch exceeds --rpc-max-batch-size.`

### PR DESCRIPTION
**Pull request by Gabriel-Trintinalia**
## PR description

Adds a `--rpc-max-batch-size` CLI option to restrict the maximum number of requests in a single RPC batch request.

Default `50`
When `-1`: unlimited - the current configuration.

When number of requests in batch > `--rpc-max-batch-size`:

- Throw a JsonRpcError.EXCEEDS_RPC_MAX_BATCH_SIZE; with type `INVALID_PARAM` and message `Number of requests in batch exceeds --rpc-max-batch-size.`

## Fixed Issue(s)
fixes #4951

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: besu/src/main/java/org/hyperledger/besu/cli/DefaultCommandValues.java [hunk 1]
```diff
@@ -86,6 +86,8 @@ public interface DefaultCommandValues {
   int DEFAULT_P2P_PEER_LOWER_BOUND = 25;
   /** The constant DEFAULT_HTTP_MAX_CONNECTIONS. */
   int DEFAULT_HTTP_MAX_CONNECTIONS = 80;
+  /** The constant DEFAULT_HTTP_MAX_BATCH_SIZE. */
+  int DEFAULT_HTTP_MAX_BATCH_SIZE = 50;
```
Comment (non-fungible-nelson) [inline besu/src/main/java/org/hyperledger/besu/cli/DefaultCommandValues.java:90]: Should mirror Geth and other clients at 1 IMO - users that leave this exposed by default will not want such a large batch size and those that need the functionality can track down this CLI option in the docs.

Comment (non-fungible-nelson) [reply besu/src/main/java/org/hyperledger/besu/cli/DefaultCommandValues.java:90]: Mirroring @macfarla comment's on #4951

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/handlers/HandlerFactory.java [hunk 2]
```diff
@@ -47,7 +48,9 @@ public static Handler<RoutingContext> jsonRpcParser() {
   }
 
   public static Handler<RoutingContext> jsonRpcExecutor(
-      final JsonRpcExecutor jsonRpcExecutor, final Tracer tracer) {
-    return JsonRpcExecutorHandler.handler(jsonRpcExecutor, tracer);
+      final JsonRpcExecutor jsonRpcExecutor,
+      final Tracer tracer,
+      final JsonRpcConfiguration configuration) {
```
Comment (gfukushima) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/handlers/HandlerFactory.java:53]: nit: suggestion of variable name to follow the same patter jsonRpcConfiguration

**Review by gfukushima — APPROVED** [review]
LGTM

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/handlers/HandlerFactory.java (hunk 2 above)
Comment (Gabriel-Trintinalia) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/handlers/HandlerFactory.java]: done

**Review by macfarla — APPROVED** [review]

**Pull request merged**